server/task_executor_schema_update_ghost_sync.go

```diff
@@ -1,26 +1,232 @@
+		}
+	}()
+
+	payload := &api.TaskDatabaseSchemaUpdateGhostSyncPayload{}
+	if err := json.Unmarshal([]byte(task.Payload), payload); err != nil {
+		return true, nil, fmt.Errorf("invalid database schema update gh-ost sync payload: %w", err)
+	}
+	return runGhostMigration(ctx, exec.l, server, task, db.Migrate, payload.Statement, payload.SchemaVersion, payload.VCSPushEvent)
+}
+
+func runGhostMigration(ctx context.Context, l *zap.Logger, server *Server, task *api.Task, migrationType db.MigrationType, statement, schemaVersion string, vcsPushEvent *vcsPlugin.PushEvent) (terminated bool, result *api.TaskRunResultPayload, err error) {
+	mi, err := preMigration(ctx, l, server, task, migrationType, statement, schemaVersion, vcsPushEvent)
+	if err != nil {
+		return true, nil, err
+	}
+	migrationID, schema, err := executeSync(ctx, l, task, mi, statement)
+	if err != nil {
+		return true, nil, err
+	}
+	return postMigration(ctx, l, server, task, vcsPushEvent, mi, migrationID, schema)
+}
+
+func executeSync(ctx context.Context, l *zap.Logger, task *api.Task, mi *db.MigrationInfo, statement string) (int64, string, error) {
+	statement = strings.TrimSpace(statement)
+
+	driver, err := getAdminDatabaseDriver(ctx, task.Instance, task.Database.Name, l)
+	if err != nil {
+		return -1, "", err
+	}
+	defer driver.Close(ctx)
+	setup, err := driver.NeedsSetupMigration(ctx)
+	if err != nil {
+		return -1, "", fmt.Errorf("failed to check migration setup for instance %q: %w", task.Instance.Name, err)
+	}
+	if setup {
+		return -1, "", common.Errorf(common.MigrationSchemaMissing, fmt.Errorf("missing migration schema for instance %q", task.Instance.Name))
+	}
+
+	executor := driver.(util.MigrationExecutor)
+
+	var prevSchemaBuf bytes.Buffer
+	if err := driver.Dump(ctx, mi.Database, &prevSchemaBuf, true); err != nil {
+		return -1, "", err
+	}
+
+	insertedID, err := util.BeginMigration(ctx, executor, mi, prevSchemaBuf.String(), statement, db.BytebaseDatabase)
+	if err != nil {
+		return -1, "", err
+	}
+	startedNs := time.Now().UnixNano()
+
+	err = executeGhost(task.Instance, task.Database.Name, statement)
+	if err != nil {
+		return -1, "", err
+	}
+
+	var afterSchemaBuf bytes.Buffer
+	if err := executor.Dump(ctx, mi.Database, &afterSchemaBuf, true); err != nil {
+		return -1, "", err
+	}
+
+	if err := util.EndMigration(ctx, l, executor, startedNs, insertedID, afterSchemaBuf.String(), mi.Database, true); err != nil {
+		l.Error("failed to update migration history record",
+			zap.Error(err),
+			zap.Int64("migration_id",
+				insertedID),
+		)
+	}
+	return insertedID, afterSchemaBuf.String(), nil
+}
+
+func executeGhost(instance *api.Instance, databaseName string, statement string) error {
+	adminDataSource := api.DataSourceFromInstanceWithType(instance, api.Admin)
+	if adminDataSource == nil {
+		return common.Errorf(common.Internal, fmt.Errorf("admin data source not found for instance %d", instance.ID))
+	}
+
+	port := 3306
+	if instance.Port != "" {
```

Comment:
Should Port be required for setting up an Instance?

Also, does migrationContext.InspectorConnectionConfig.Key.Port convert default 0 to 3306? If not, maybe setting default 3306 should be part of newMigrationContext().